GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -234,22 +237,41 @@ bool IsOverride(MethodInfo m)
         }
 
         /// <summary>
-        /// Toggles filtering mode on or off to the git refs present in the left panel depending on the app's global filtering rules .
-        /// These rules include: show all branches / show current branch / show filtered branches, etc.
+        /// FormBrowse refreshing the side panel when refreshing the grid.
         /// </summary>
         /// <param name="isFiltering">
         ///  <see langword="true"/>, if the data is being filtered; otherwise <see langword="false"/>.
         /// </param>
-        /// <param name="getRefs">Function to get refs.</param>
         /// <param name="forceRefresh">Refresh may be required as references may have been changed.</param>
-        public void Refresh(bool isFiltering, bool forceRefresh, Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs)
+        /// <param name="getRefs">Function to get refs.</param>
+        /// <param name="getStashRevs">Lazy accessor for stash commits.</param>
+        public void RefreshWhenLoading(bool isFiltering, bool forceRefresh, Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs, Lazy<IReadOnlyCollection<GitRevision>> getStashRevs)
```

Comment:
I find `RefreshWhenLoading` and `RefreshAfterLoaded` confusing. Without reading the xml-docs I couldn't quite figure out what these two do...
I'm having troubles with the meaning of "refreshing while data is loading". `RevisionGrid.RevisionsLoading` event is raised _before_ data is loaded, and `RevisionGrid.RevisionsLoaded` is raised _after_ the data has been loaded and displayed. According to the comment, we have added `RevisionGrid.RevisionsLoading` event specifically to refresh the left panel.

Was there a specific reason why you thought the two methods were necessary? Couldn't we have the single method with an additional flag to indicate whether the data _is being_ or _has been_ loaded?

Reply:
Before it was Refresh() in many situations which was more confusing.
The first call is before the revision grid is available. The data to add is there, but not if a revision is visible or not, which is available after loading. RefreshAfterLoaded() do not need any parameters right now, it just updates the visibility.

The previous commit updated stash info after loading only. That strategy could be used for all 'commit' (branches, tags, stashes but not submodules), but then updating of will be delayed.

So I prefer two methods, possibly with updated arguments.

Reply:
Renaming to `RefreshRevisionsLoading` and `RefreshRevisionsLoaded` so similar to the events.
Keeping the arguments the same, as isFiltering is needed after loading (to skip updating visibility).

Reply:
I renamed the repoObjectsTree method `Refresh` to `RevisionGrid.RevisionsLoading -> RefreshRevisionsLoading` and added `RevisionGrid.RevisionsLoaded -> RefreshRevisionsLoaded`, called from .
The call to RepoObjectTree is now wrapped in `FormBrowse.RefreshLeftPanel`, but keeping the changed names.